### PR TITLE
Convert battery icon to vue component

### PR DESCRIPTION
**Pull request by chmelevskij**
Convert battery icon logic to self contained vue component.

Screenshot 2022-12-30 at 20 25 52
Screenshot 2022-12-30 at 20 25 46
Screenshot 2022-12-30 at 20 25 41

### CODE REVIEW

**Comment by chmelevskij** [comment]
Seems to be working, just need to fix one little issue where is shows it's empty battery when connected to usb (might be issue which was there before as well, but need to do some manual verification)

File: src/components/quad-status/BatteryIcon.vue
Comment (McGiverGim) [inline src/components/quad-status/BatteryIcon.vue:13]: Only as suggestion... now that you have refactored it... maybe we can get rid of this, now that the code is clean...

We need to call this MSP:
https://github.com/betaflight/betaflight-configurator/blob/d0d3141b7202004575dfd9f2a162b21dc96fbfb7/src/js/msp/MSPHelper.js#L307-L318

and pass the FC.BATTERY_STATE.batteryState to the component. The value will be:

https://github.com/betaflight/betaflight/blob/025ee87a7aca068e3659fd066b8a9afbed123361/src/main/sensors/battery.h#L84-L90

It can be null or undefined (I'm not too sure) if the firmware is old, in this case we need to use the same code than now.

Comment (chmelevskij) [reply src/components/quad-status/BatteryIcon.vue:13]: Will grab a look at it 😉 thanks for suggestion

File: src/components/quad-status/BatteryIcon.vue
Comment (chmelevskij) [inline src/components/quad-status/BatteryIcon.vue:64]: If someone has firmware ready with `batteryState` would be good to check.

File: src/css/theme.css
Comment (chmelevskij) [inline src/css/theme.css:29]: Moving out for better visibility and easier usage with no less.

File: src/components/quad-status/BatteryIcon.vue
Comment (haslinghuis) [reply src/components/quad-status/BatteryIcon.vue:64]: Can check in a few hours

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
Testing now but Configurator does not start?

image

**Comment by chmelevskij** [comment]
@haslinghuis fixed now

File: src/components/quad-status/BatteryIcon.vue
Comment (haslinghuis) [reply src/components/quad-status/BatteryIcon.vue:64]: Working now. Tested with and without battery. Icon is live.

File: src/components/quad-status/BatteryIcon.vue
Comment (haslinghuis) [inline src/components/quad-status/BatteryIcon.vue:57]: space after if

File: src/components/quad-status/BatteryIcon.vue
Comment (haslinghuis) [reply src/components/quad-status/BatteryIcon.vue:13]: Configurator now only supports >= API_VERSION_1_42

**Review by haslinghuis — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
